### DISCUSSION TITLE
Question: `useQuery` and async

### DISCUSSION THREAD

**Discussion by netheril96** [blitz-js/blitz#1002] [General] [answered]
The definition of [useQuery](https://blitzjs.com/docs/use-query) seemingly converts asynchronous functions into a synchronous one. How does that work? Does it cause blocking in frontend? How to access the promise instead?

**Comment by flybayer** [comment, marked as answer]
Good question! The short answer is because it use React Suspense (concurrent mode).

So while the network request is inflight, it triggers suspense which wall cause the fallback component on the nearest `<Suspense fallback={/*stuff*/}>` component.

So the component render basically stops at the `useQuery` call until the promise is resolved. Once resolved, then the component will fully render.

Does that makes sense?

**Reply by netheril96** [reply]
It does make sense, but it seems too magical for me. What if I just want to call the query as a normal async function? Does blitz rewrite that into a network call, just like with mutations (https://blitzjs.com/docs/mutation-usage)?

**Reply by flybayer** [reply]
Yeah you can totally do that too. You can manually call queries just like mutations.

```ts
const res = await getProducts()
```

**Reply by flybayer** [reply]
`useQuery` definitely does a ton of stuff for you that can make it seem a bit magical. It handles caching, cache invalidation, auto refetching if you leave the tab and come back to it, etc. It also gives you ability to do stuff like polling.
